src/engine/GroupBy.cpp

```diff
@@ -908,25 +886,19 @@ void GroupBy::substituteAllAggregates(
 
 // _____________________________________________________________________________
 void GroupBy::createResultFromHashMap(
-    IdTable* result,
-    const ad_utility::HashMapWithMemoryLimit<KeyType, ValueType>& map,
+    IdTable* result, const HashMapAggregationData& aggregationData,
     std::vector<HashMapAliasInformation>& aggregateAliases,
-    const std::vector<std::vector<AverageAggregationData>>& aggregationData,
     LocalVocab* localVocab) {
-  // Sort by groupBy column
-  std::vector<ValueId> sortedKeys;
-  for (const auto& val : map) {
-    sortedKeys.push_back(val.first);
-  }
-  std::ranges::sort(sortedKeys);
-
   // Create result table, filling in the group values, since they might be
   // required in evaluation
-  // TODO<C++23> use views::enumerate
-  size_t rowIndex = 0;
+  auto sortedKeys = aggregationData.getSortedGroupColumn();
+  size_t numberOfGroups = aggregationData.getNumberOfGroups();
+  result->resize(numberOfGroups);
+
+  auto targetIterator = result->getColumn(0).begin();
   for (const auto& val : sortedKeys) {
-    result->emplace_back();
-    (*result)(rowIndex++, 0) = val;
+    *targetIterator = val;
+    ++targetIterator;
```

Comment:
This whole block can be `std::ranges::copy(sortedKeys, result->getColumn(0).begin())`